### DISCUSSION TITLE
Unpaused Midi-Notes

### DISCUSSION THREAD

**Discussion by Tutorius** [DISTRHO/Cardinal#331] [Q&A] [answered]
Hi,
i use Cardinal as a modular synth on a linux-machine, use carla as DAW and MusE as sequencer. I use the lv2-Version of Cardinal.
I use self created midi-files, and sometime downloaded ones. I have the problem that the Midi-Input-Module inside Cardinal does not create a gate-signal if notes are played without a "gap".
On a keyboard, there is no problem, you will always create a gap betwenn the note and the following note.
But when editing notes by hand on the seqeuncer, thery will be no gaps, the notw will magnetical end on the beginning of the next note.
And the ADSR-units will not get a gate-impulse in these cases.
What can i do to get my notes played correctly? Is there a module outside Cardinal (perhaps inside Carla) that i can use to "shorten" the notes? Or is there an other Midi-Module to be used inside Cardinal?

Thanks fpr your help.

**Comment by falkTX** [comment, marked as answer]
This is just how Rack works. gates never stop for notes if there is not enough of a pause between them.
Apparently we are meant to use the Retrigger CV source for this.

I find this to be quite the bad design, but for compatibility reasons the current Host MIDI module works the same way as the official VCV Rack one.
At some point I will do a "Host MIDI v2" module that handles this in hopefully a better way.

**Reply by Tutorius** [reply]
Thanks for your answer. Found the time today to build something in Carla/Cardinal. I have "ignored" the Retrigger-thing since your post :)
I achieved well working Synths with "problematic" midi-files by X-OR ing the outputs of "Gate" and "Retrigger". Seems to work with all kinds of midi-situations.
Thanks for your help.
